pallets/gear/src/tests.rs

```diff
@@ -522,13 +524,39 @@ fn delayed_send_user_message_payment() {
                 .saturating_mul(CostsPerBlockOf::<Test>::dispatch_stash()),
         );
 
+        let reserve_for_fee = GasPrice::gas_price(
+            CostsPerBlockOf::<Test>::reserve_for()
+                .saturating_mul(CostsPerBlockOf::<Test>::dispatch_stash()),
+        );
+
         // Gas should be reserved until message is holding.
         assert_eq!(Balances::reserved_balance(&USER_1), delay_holding_fee);
+        let free_balance = Balances::free_balance(&USER_1) + Balances::reserved_balance(&USER_1);
 
-        // Run blocks to release message.
-        run_to_block(delay + 3, None);
+        // Run blocks before sending message.
+        run_to_block(delay + 2, None);
+
+        let delayed_id = MessageId::generate_outgoing(proxy_msg_id, 0);
+
+        // Check that delayed task was created
+        assert!(TaskPoolOf::<Test>::contains(
+            &(delay + 3),
+            &ScheduledTask::SendUserMessage {
+                message_id: delayed_id,
+                to_mailbox: false
+            }
+        ));
+
+        // Mailbox should be empty
+        assert!(MailboxOf::<Test>::is_empty(&USER_2));
+
+        run_to_next_block(None);
 
```

Comment:
assert has event of sending but not included in mailbox once executed delayed task

Reply:
done